MSVC/gen-build-config.bat

```diff
@@ -182,74 +196,65 @@ EXIT /B %ERRORLEVEL%
 	::   2 * nn + 1 if the working tree was dirty ('-WIP')
 	::   65535 if unknown ('X')
 	:: - 4rd 16-bit number: 1st 4 hex digits of cccccccc (commit SHA1)
-	IF /I "%$MINOR_VERSION_COMMITS%" == "X" (
+	IF /I "%$MINOR_VERSION_COMMITS%" =="X" (
 		SET /A $MIV3=65535
-	) ELSE IF EXIST $MINOR_VERSION_WIP (
+	)
+	IF EXIST $MINOR_VERSION_WIP (
 		SET /A $MIV3=2*%$MINOR_VERSION_COMMITS%+1
 	) ELSE (
 		SET /A $MIV3=2*%$MINOR_VERSION_COMMITS%
 	)
-	IF /I "%$MINOR_VERSION_COMMITS%" == "X" (
+	IF /I "%$MINOR_VERSION_COMMITS%" =="X" (
 		SET /A $MIV4=0
 	) ELSE (
-		SET /A $MIV4=0x%$MINOR_VERSION_SHA1:~0,4%
+		SET $TMP =
+		SET /A $MIV4=0x%$MINOR_VERSION_SHA1:~1,4%
+
 	)
-	SET $BINARY_VERSION=0,%$MAJOR_VERSION:~2,6%,%$MIV3%,%$MIV4%
+	SET $VERSION_BINARY=0,%$MAJOR_VERSION:~2,6%,%$MIV3%,%$MIV4%
 	EXIT /B 0
 
 :: Report version info to the console
 :REPORT_VERSION
-	ECHO Minetestmapper version information:
+	ECHO Minetestmapper version information^:
 	ECHO.
-	IF /I "%$MINOR_VERSION_COMMITS%" == "X" (
-		ECHO WARNING: git executable not found, or no git tree found.
+	IF /I "%$MINOR_VERSION_COMMITS%" =="X" (
+		ECHO WARNING^: git executable not found, or no git tree found.
 		ECHO          No exact version information could be obtained.
 		ECHO.
 	)
-	ECHO Full version:                   %$FULL_VERSION%
-	ECHO Major version:                  %$MAJOR_VERSION%
-	ECHO Minor version:                  %$MINOR_VERSION%
-	IF /I NOT "%$MINOR_VERSION_COMMITS%" == "X" (
-		ECHO Commits since full version:     %$MINOR_VERSION_COMMITS%
-		ECHO Latest commit SHA1 id (short):  %$MINOR_VERSION_SHA1%
-		IF EXIST $MINOR_VERSION_WIP (
-			ECHO Working tree is clean:          NO
+	ECHO Full version^:                  %$FULL_VERSION%
+	ECHO Major version^:                  %$MAJOR_VERSION%
+	ECHO Minor version^:                  %$MINOR_VERSION%
+	
+	IF /I NOT "%$MINOR_VERSION_COMMITS%"=="X" (
+		ECHO Commits since full version^:     %$MINOR_VERSION_COMMITS%
+		ECHO Latest commit SHA1 id ^(short^)  %$MINOR_VERSION_SHA1%
+		IF EXIST %$MINOR_VERSION_WIP% (
+			ECHO Working tree is clean          NO
 		) ELSE (
-			ECHO Working tree is clean:          YES
+			ECHO Working tree is clean          YES
 		)
 	)
-	ECHO Digital version:                %$BINARY_VERSION%
+	ECHO Digital version^:                %$BINARY_VERSION%
 	exit /B 0
 	
 :: Create the build file
 :WRITE_BUILDCONFIG
-	SET $INDENT=
+	SET $INDENT=""
 	IF NOT EXIST %$PROJECTDIR%\%$BUILDCONFIG_TEMPLATE% (
-		ECHO %$MYNAME%: Error: template file (%$PROJECTDIR%\%$BUILDCONFIG_TEMPLATE%) not found.
+		ECHO %$MYNAME%^: Error^: template file ^(%$PROJECTDIR%\%$BUILDCONFIG_TEMPLATE%^) not found.
 		exit /B 1
 	)
-	IF %$TEST% EQU 1 (
-		ECHO The build file (build_config.h) contents would be:
-		CALL "SET $INDENT=    "
-	) ELSE IF %$VERBOSE% EQU 1 (
-		ECHO ---- Generating %$PROJECTDIR%\%$BUILDCONFIG%
+	IF %$TEST%==1 (
+		ECHO The build file ^(build_config.h^) contents would be^:
+		SET $INDENT=    
```

Comment:
I looked it up again, and I tested it n my mother's computer, and it should have been:
```CALL SET "$INDENT=    "```

Fixed.